v-next/hardhat-node-test-coverage/src/collector.ts

```diff
@@ -0,0 +1,44 @@
+import type { CoverageReport } from "hardhat/types/coverage";
+import type { NetworkHooks } from "hardhat/types/hooks";
+
+import { randomUUID } from "node:crypto";
+import path from "node:path";
+import { after, before } from "node:test";
+
+import { assertHardhatInvariant } from "@nomicfoundation/hardhat-errors";
+import { writeJsonFile } from "@nomicfoundation/hardhat-utils/fs";
+import hre from "hardhat";
+
+const report: CoverageReport = {
+  markerIds: [],
+};
+
+before(() => {
+  const networkHooks: Partial<NetworkHooks> = {
+    newConnection: async (
+      context,
+      networkName,
+      chainType,
+      networkConfigOverride,
+      next,
+    ) => {
+      // TODO: Define networkConfigOverride.observability.codeCoverage.onCollectedCoverageCallback
```

Comment:
Note that these shouldn't be exposed as part of the config. The Hardhat config of EDR networks shouldn't match the EDR internal config format. They will evolve independently.

Reply:
Having written that, I think this approach won't work. Because this handler won't be able to access that internal config field.

Reply:
I see your point! In that case, I think modifying the EDR API so that we pull the coverage rather than register a callback might in fact be a better option.

That way, we could revert the modification of new connection hook signature, add a small indicator in network manager that controls whether EDR networks should be created with coverage enabled or not, and add a get call for the coverage data here once the tests are complete.

Reply:
Alternatively, as you're saying, we could make network manager aware of the coverage global option and give it a responsibility to act as an intermediary in handling the callbacks.